federation/pkg/kubefed/init/init.go

```diff
@@ -442,7 +454,16 @@ func createNamespace(clientset client.Interface, federationName, namespace strin
 	return clientset.Core().Namespaces().Create(ns)
 }
 
-func createService(cmdOut io.Writer, clientset client.Interface, namespace, svcName, federationName, apiserverAdvertiseAddress string, apiserverServiceType v1.ServiceType, dryRun bool) (*api.Service, []string, []string, error) {
+func createService(cmdOut io.Writer, clientset client.Interface, namespace, svcName, federationName, apiserverAdvertiseAddress string, apiserverPort int32, apiserverServiceType v1.ServiceType, dryRun bool) (*api.Service, []string, []string, error) {
+	port := api.ServicePort {
+			Name:       "https",
+			Protocol:   "TCP",
+			Port:       443,
+			TargetPort: intstr.FromString(apiServerSecurePortName),
+	}
+	if apiserverServiceType == v1.ServiceTypeNodePort {
+		port.NodePort = apiserverPort
```

Comment:
When nodeport is specified kubernetes will appoint a random port as described here: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport

Random port assignment happens within kubernetes. Even if we wanted to do a random port assignment in the context of kubefed init we would not know the port range specified by the cluster admin.

Thank you for the quick review.